### PR TITLE
Feature: SNS msg delivery to Firehose endpoint

### PR DESCRIPTION
**Pull request by pinzon**
This PR addresses issue #5602. Allows the SNS to deliver messages to Firehose subscriptions.

Must be merged after spulec/moto#5303 has been integrated. If it takes too long I guess I'll merge it into the Moto-ext fork.

### CODE REVIEW

**Comment by thrau** [comment]
hey @pinzon, thanks for picking this up! any chance you could provide a test for this? ideally something that's AWS validated. thanks!

**Comment by pinzon** [comment]
@thrau of course, I just opened to this draft to indicate that the issue is being work on.

**Comment by thrau** [comment]
my bad! looks like i missed the draft state. Carry on! 😁

File: tests/integration/test_sns.py [hunk 1]
```diff
@@ -2113,3 +2115,115 @@ def test_publish_to_gcm(self, sns_client):
...
+        retries = 5
+        sleep = 1
+        sleep_before = 0
+        if is_aws_cloud():
+            retries = 25
+            sleep = 10
+            sleep_before = 10
+
+        retry(validate_content, retries=retries, sleep_before=sleep_before, sleep=sleep)
+        firehose_client.delete_delivery_stream(DeliveryStreamName=stream_name)
```
Comment (bentsku) [inline tests/integration/test_sns.py:2229]: nit: should we manage the delivery stream in a fixture as well, to assure the cleanup of the resource? We could use the `wait_for_delivery_stream` in the fixture as well.

File: tests/integration/test_sns.py [hunk 2]
```diff
@@ -2113,3 +2115,115 @@ def test_publish_to_gcm(self, sns_client):
...
+        wait_for_delivery_stream_ready,
+        sns_create_topic,
+        sns_subscription,
+    ):
+        role_name = f"test-role-{short_uid()}"
+        stream_name = f"test-stream-{short_uid()}"
+        bucket_name = f"test-bucket-{short_uid()}"
+        topic_name = f"test_topic_{short_uid()}"
+
+        subscriptionRoleArn = f"arn:aws:iam::{get_aws_account_id()}:role/{role_name}"
```
Comment (bentsku) [inline tests/integration/test_sns.py:2137]: nit: this is reassigned later on and not used, could we delete it?

**Review by bentsku — APPROVED** [review]
LGTM! Minor nits concerning the test cleanup, but otherwise all good! 👍
Thanks a lot for adding this feature! Sorry it took so long with the upstream change, thanks for fixing it there as well.

**Comment by pinzon** [comment]
Thanks @bentsku for the review. I just uploaded the changes. :smiley:

File: localstack/testing/pytest/fixtures.py [hunk 3]
```diff
@@ -1441,6 +1458,25 @@ def _create_role_and_policy(**kwargs):
         iam_client.delete_role(RoleName=role_name)
 
 
+@pytest.fixture
+def firehose_create_delivery_stream(firehose_client, wait_for_delivery_stream_ready):
+    delivery_streams = {}
+
+    def _create_delivery_stream(**kwargs):
+        if "DeliveryStreamName" in kwargs:
```
Comment (bentsku) [inline localstack/testing/pytest/fixtures.py:1466]: nit: should it be `not in`?

**Review by bentsku — APPROVED** [review]
Thanks for the updated change! Love the fixtures! 🎉 just a minor nit concerning the logic for name, but otherwise all good. Thank you!

File: localstack/testing/pytest/fixtures.py (hunk 3 above)
Comment (pinzon) [reply localstack/testing/pytest/fixtures.py]: @bentsku Thanks for catching this, I will be more careful.

Comment (bentsku) [reply localstack/testing/pytest/fixtures.py]: No worries, that's what PR reviews are for 😁

**Pull request merged**